### ISSUE TITLE
Command line option -o does not produce output with -g option

### ISSUE THREAD

**Issue by Fernal73** [checkstyle/checkstyle#6934]
https://checkstyle.org/cmdline.html#Command_line_usage

>-g,--generate-xpath-suppression - generate to output a suppression.xml to use to suppress all violations from user's config. 

```bash
java -cp {classpath/jarfile}\
 com.puppycrawl.tools.checkstyle.Main\
 -g\
 -c {config file}\
 -p {properties file}\
 -o cs.errors\
 {list of java sources} 

```
Detected:

Suppressions xpath xml output is generated if checkstyle rules under Tree Walker module are violated. 
Argument '-o cs.errors'  is ignored, by design, but it would be better to respect `-o`. 

---------------
Expected:
Error file as specified in option -o must be output of `-g`. 

--------------
Versions run against : 8.22, 8.23.

**Comment by romani** [comment]
Fix is merged

**Comment by romani** [comment]
@strkkk , please update xdoc, it has

>-g,--generate-xpath-suppression - generate to output a suppression.xml to use to suppress all violations from user's config.

**Comment by strkkk** [comment]
@romani what is wrong with it?

**Comment by Fernal73** [comment]
The output also includes a line that informs the user of the number of errors found. That should not be part of the xml output. That will help while redirecting the xml to a file via say a 'tee' command. As it stands, that additional line forms part of the xml output.

**Comment by strkkk** [comment]
> The output also includes a line that informs the user of the number of errors found. That should not be part of the xml output

I think you can create new issue for it

**Comment by Fernal73** [comment]
I'm unaware if it has been fixed with the above. If not, yes, a new issue will have to be raised.

**Comment by romani** [comment]
>what is wrong with it?

In doc:
>suppression.xml 

Should be:
`suppression xml` as we will generate file with name defined by `-o`, so we do not know name, it is not hatdcoded anymore

**Comment by romani** [comment]
doc fix is merged.

**Comment by Fernal73** [comment]
The documentation could also be updated as to how to run Checkstyle as an ant java task and set the -g parameter since the existing Ant task for Checkstyle doesn't allow that option to be set.

**Comment by romani** [comment]
> ow to run Checkstyle as an ant java task and set the -g parameter

it is not for ANT, such option will never be in ANT.

**Comment by Fernal73** [comment]
Exactly. But it can be run as an Ant Java task if the user wishes to with all the command line parameters. I'm not referring to the Checkstyle ant task. That's separate.

**Comment by romani** [comment]
let postpone request creation, ant task is heavily underused, user use maven/gradle more then ANT.

**Comment by Fernal73** [comment]
That's fine with me. I can always run my batch file through Ant if needed.

**Comment by Fernal73** [comment]
```xml
 <target name="checkstyleg">
    <fileset dir="${basedir}"
             id="javasrcs">
      <include name="**/*.java" />
    </fileset>
    <pathconvert property="sources"
                 refid="javasrcs"
                 pathsep=" " />
            <java classname="${cs.main.class}" output="${ant.project.name}-xpath.xml"
              logError="true">
      <arg line="-c ../${cs.config} -p ${cs.properties} -o ${cs.output} -g ${sources}" />
      <classpath>
        <pathelement path="{checkstyle classpath} " />
        <pathelement path="${java.class.path}" />
      </classpath>
    </java>
  </target>
```
cs.main.class -> com.puppycrawl.tools.checkstyle.Main
cs.properties -> properties file
cs.output -> output file

https://checkstyle.sourceforge.io/cmdline.html

This may help.. 

Regards, 
Linus.
